test/OpenTelemetry.Collector.AspNet.Tests/HttpInListenerTests.cs

```diff
@@ -0,0 +1,168 @@
+﻿// <copyright file="HttpInListenerTests.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Web;
+using System.Web.Routing;
+using Moq;
+using OpenTelemetry.Trace;
+using OpenTelemetry.Trace.Configuration;
+using OpenTelemetry.Trace.Export;
+using Xunit;
+
+namespace OpenTelemetry.Collector.AspNet.Tests
+{
+    public class HttpInListenerTests : IDisposable
+    {
+        private readonly FakeAspNetDiagnosticSource fakeAspNetDiagnosticSource;
+
+        public HttpInListenerTests()
+        {
+            this.fakeAspNetDiagnosticSource = new FakeAspNetDiagnosticSource();
+        }
+
+        public void Dispose()
+        {
+            this.fakeAspNetDiagnosticSource.Dispose();
+        }
+
+        [Theory]
+        [InlineData("https://localhost:443/", 0, null)]
```

Comment:
Please add a test case where port is not specified. I think we should default to 80 then.

Reply:
On next commit I updated the InlineData to have http & https with no ports, 80 & 443 specified, and two custom ports specified over http & https.